node_package/src/clientStartup.js

```diff
@@ -0,0 +1,84 @@
+import ReactDOM from 'react-dom';
+import handleError from './handleError';
+import createReactElement from './createReactElement';
+import isRouterResult from './isRouterResult';
+
+const REACT_ON_RAILS_COMPONENT_CLASS_NAME = 'js-react-on-rails-component';
+const turbolinksInstalled = (typeof Turbolinks !== 'undefined');
+
+function forEachComponent(fn) {
+  const els = document.getElementsByClassName(REACT_ON_RAILS_COMPONENT_CLASS_NAME);
+  for (let i = 0; i < els.length; i++) {
+    fn(els[i]);
+  }
+}
+
+function render(el) {
+  const componentName = el.getAttribute('data-component-name');
+  const domId = el.getAttribute('data-dom-id');
+  const props = JSON.parse(el.getAttribute('data-props'));
+  const trace = JSON.parse(el.getAttribute('data-trace'));
+  const generatorFunction = JSON.parse(el.getAttribute('data-generator-function'));
+  const expectTurboLinks = JSON.parse(el.getAttribute('data-expect-turbo-links'));
```

Comment:
who is putting this attribute here, Rails or react_on_rails? Either way, checking for this is a cool idea. Belt and suspenders!